pkg/service/model_config.go

```diff
@@ -121,6 +123,11 @@ type Action struct {
 	Return     *Return
 }
 
+type Future struct {
+	Delay time.Duration // e.g. "12h" or "10s"
+	Action
```

Comment:
```suggestion
	Action

	Delay time.Duration // e.g. "12h" or "10s"
```

Can we verify YAML can be on the root level of `Future` and be read into all fields? I want to ensure that unmarshal works.

Reply:
the YAML needs to have it like this, with the "action" tag.
```
      future:
        delay: "12h"
        action:
          correction:
            code: "C01"
            data: "744567899"
```

Reply:
Why? We could make it cleaner and consolidate the two. `copy` isn't valid either.

Reply:
so then maybe in the `Future` struct, it doesn't nest `Action`, it just holds the valid elements of `Action` (being `*Return` and `*Correction`)

Reply:
see comment below